lib/daru/io/base.rb

```diff
@@ -1,15 +1,21 @@
+require 'daru'
+
 module Daru
   module IO
     class Base
-      def initialize(binding)
-        args = method(__method__).parameters.map do |_, name|
-          [name, binding.local_variable_get(name.to_s)]
-        end.to_h
-
-        args.each do |k, v|
-          instance_variable_set("@#{k}", v)
-          define_singleton_method(k) { instance_variable_get("@#{k}") }
-        end
+      def optional_gem(dependency, version=nil, requires: nil)
+        gem dependency, version
+        require requires || dependency
+      rescue LoadError
+        statement =
```

Comment:
Is this good enough? That is, should the rescued `LoadError` raise a `LoadError` again with a better error message or are there any better alternatives (like a simple `raise`, or maybe even custom error class?)